src/top_model/main.cpp

```diff
@@ -0,0 +1,261 @@
+#include <iostream>
+#include <chrono>
+#include <algorithm>
+#include <string>
+#include <cadmium\modeling\coupling.hpp>
+#include <cadmium\modeling\ports.hpp>
+#include <cadmium\modeling\dynamic_model_translator.hpp>
+#include <cadmium\concept\coupled_model_assert.hpp>
+#include <cadmium\modeling\dynamic_coupled.hpp>
+#include <cadmium\modeling\dynamic_atomic.hpp>
+#include <cadmium\engine\pdevs_dynamic_runner.hpp>
+#include <cadmium\logger\tuple_to_ostream.hpp>
+#include <cadmium\logger\common_loggers.hpp>
+
+#include "../../include/atomics/subnet_cadmium.hpp"
+#include "..\..\lib\vendor\NDTime.hpp"
+#include "..\..\lib\vendor\iestream.hpp"
+#include "..\..\include\data_structures\message.hpp"
+#include "..\..\include\atomics\sender_cadmium.hpp"
+#include "..\..\include\atomics\receiver_cadmium.hpp"
+
+using namespace std;
+using hclock=chrono::high_resolution_clock;
+using TIME = NDTime;
+
+/***** SETING INPUT PORTS FOR COUPLEDs *****/
+struct inp_control : public cadmium::in_port<Message_t>{};
+struct inp_1 : public cadmium::in_port<Message_t>{};
+struct inp_2 : public cadmium::in_port<Message_t>{};
+
+/***** SETING OUTPUT PORTS FOR COUPLEDs *****/
+struct outp_ack : public cadmium::out_port<Message_t>{};
+struct outp_1 : public cadmium::out_port<Message_t>{};
+struct outp_2 : public cadmium::out_port<Message_t>{};
+struct outp_pack : public cadmium::out_port<Message_t>{};
+
+/********************************************/
+/****** APPLICATION GENERATOR *******************/
+/********************************************/
+
+template<typename T>
+class ApplicationGen : public iestream_input<Message_t,T> {
+	public:
+		ApplicationGen() = default;
+		ApplicationGen(const char* file_path) :
+		iestream_input<Message_t,T>(file_path) {}
```

Comment:
A suggestion you can keep 4 indentations so that it looks like it is in continuation - line 46 and there are more stuff line 97 and so on.
Thank you

Reply:
Please download the files to your local and check the indentations.
Its looking perfect in notepad++ or other editor with 4 spaces

Reply:
Requested changes has been updated.can you please look and let me know.
As discussed after applied specific changes related to space(tab),still in git account files are not coming properly.
So can you please look into your local system and let me know how its looks.
So i will update based on that to prof. as well.